WcaOnRails/spec/factories/competitions.rb

```diff
@@ -44,5 +44,10 @@
       with_delegate
       isConfirmed true
     end
+
+    trait :confirmed_and_visible do
```

Comment:
I can confirm that FactoryGirl seems to do the right thing here. With the following changes:

```
~/gitting/worldcubeassociation.org/WcaOnRails @kaladin> git diff
diff --git a/WcaOnRails/spec/factories/competitions.rb b/WcaOnRails/spec/factories/competitions.rb
index 3417731..c8579a6 100644
--- a/WcaOnRails/spec/factories/competitions.rb
+++ b/WcaOnRails/spec/factories/competitions.rb
@@ -18,7 +18,7 @@ FactoryGirl.define do
     venue "My backyard"
     venueAddress "My backyard street"
     website "https://www.worldcubeassociation.org"
-    showAtAll true
+    #<<<showAtAll true
 
     guests_enabled true
 
@@ -44,5 +44,10 @@ FactoryGirl.define do
       with_delegate
       isConfirmed true
     end
+
+    trait :visible do
+      with_delegate
+      showAtAll true
+    end
   end
 end
~/gitting/worldcubeassociation.org/WcaOnRails @kaladin> 
```

I see the following behavior:

```
irb(main):004:0> FactoryGirl.create(:competition, :confirmed).delegates.length
   (0.2ms)  BEGIN
  SQL (0.2ms)  INSERT INTO `Persons` (`id`, `name`, `countryId`, `gender`, `year`, `month`, `day`) VALUES ('2003FLEI03', 'Brittany Christiansen I', 'USA', 'm', 1966, 4, 4)
   (11.3ms)  COMMIT
   (0.1ms)  BEGIN
  Person Load (0.3ms)  SELECT  `Persons`.* FROM `Persons` WHERE `Persons`.`id` = '2003FLEI03' LIMIT 1
  User Exists (0.2ms)  SELECT  1 AS one FROM `users` WHERE `users`.`email` = BINARY 'ken_streich@farrell.biz' LIMIT 1
  User Load (0.2ms)  SELECT  `users`.* FROM `users` WHERE `users`.`wca_id` = '2003FLEI03' LIMIT 1
  User Load (0.2ms)  SELECT `users`.* FROM `users` WHERE `users`.`wca_id` = '2003FLEI03'
  SQL (0.2ms)  INSERT INTO `users` (`name`, `email`, `country_iso2`, `gender`, `dob`, `encrypted_password`, `wca_id`, `delegate_status`, `confirmed_at`, `created_at`, `updated_at`) VALUES ('Brittany Christiansen I', 'ken_streich@farrell.biz', 'US', 'm', '1966-04-04', '$2a$04$4ogO1bHMW2QlXaRzPiHEquk2JkT.NUkV1OIbji5kfHQZP7MlKrW9O', '2003FLEI03', 'delegate', '2016-05-17 03:02:31', '2016-05-17 03:02:31', '2016-05-17 03:02:31')
   (9.6ms)  COMMIT
   (0.1ms)  BEGIN
  Competition Exists (0.3ms)  SELECT  1 AS one FROM `Competitions` WHERE `Competitions`.`id` = BINARY 'FooComp32015' LIMIT 1
  SQL (0.3ms)  INSERT INTO `Competitions` (`name`, `cityName`, `countryId`, `information`, `eventSpecs`, `venue`, `venueAddress`, `website`, `registration_open`, `registration_close`, `isConfirmed`, `year`, `month`, `day`, `endMonth`, `endDay`, `cellName`, `id`) VALUES ('Foo Comp 3 2015', 'San Francisco', 'USA', 'Information!', '333 333oh', 'My backyard', 'My backyard street', 'https://www.worldcubeassociation.org', '2016-05-03 03:01:23', '2016-05-10 03:01:23', 1, 2015, 5, 17, 5, 17, 'Foo Comp 3 2015', 'FooComp32015')
  SQL (0.2ms)  INSERT INTO `competition_delegates` (`competition_id`, `delegate_id`, `created_at`, `updated_at`) VALUES ('FooComp32015', 3, '2016-05-17 03:02:31', '2016-05-17 03:02:31')
  User Load (0.3ms)  SELECT `users`.* FROM `users` INNER JOIN `competition_organizers` ON `users`.`id` = `competition_organizers`.`organizer_id` WHERE `competition_organizers`.`competition_id` = 'FooComp32015'
  SQL (0.2ms)  DELETE FROM `competition_organizers` WHERE `competition_organizers`.`competition_id` = 'FooComp32015' AND (1=1)
  SQL (0.2ms)  DELETE FROM `competition_delegates` WHERE `competition_delegates`.`competition_id` = 'FooComp32015' AND (`competition_delegates`.`delegate_id` != 3)
  SQL (0.1ms)  UPDATE `Results` SET `Results`.`competitionId` = 'FooComp32015' WHERE `Results`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `Preregs` SET `Preregs`.`competitionId` = 'FooComp32015' WHERE `Preregs`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `Scrambles` SET `Scrambles`.`competitionId` = 'FooComp32015' WHERE `Scrambles`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `CompetitionsMedia` SET `CompetitionsMedia`.`competitionId` = 'FooComp32015' WHERE `CompetitionsMedia`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `competition_delegates` SET `competition_delegates`.`competition_id` = 'FooComp32015' WHERE `competition_delegates`.`competition_id` = ''
  SQL (0.1ms)  UPDATE `competition_organizers` SET `competition_organizers`.`competition_id` = 'FooComp32015' WHERE `competition_organizers`.`competition_id` = ''
   (16.0ms)  COMMIT
=> 1
irb(main):005:0> FactoryGirl.create(:competition, :visible).delegates.length
   (0.1ms)  BEGIN
  SQL (0.2ms)  INSERT INTO `Persons` (`id`, `name`, `countryId`, `gender`, `year`, `month`, `day`) VALUES ('2003FLEI04', 'Tabitha Schultz DDS', 'USA', 'm', 1966, 4, 4)
   (11.2ms)  COMMIT
   (0.1ms)  BEGIN
  Person Load (0.3ms)  SELECT  `Persons`.* FROM `Persons` WHERE `Persons`.`id` = '2003FLEI04' LIMIT 1
  User Exists (0.2ms)  SELECT  1 AS one FROM `users` WHERE `users`.`email` = BINARY 'narciso_bednar@gleasonkozey.org' LIMIT 1
  User Load (0.2ms)  SELECT  `users`.* FROM `users` WHERE `users`.`wca_id` = '2003FLEI04' LIMIT 1
  User Load (0.2ms)  SELECT `users`.* FROM `users` WHERE `users`.`wca_id` = '2003FLEI04'
  SQL (0.2ms)  INSERT INTO `users` (`name`, `email`, `country_iso2`, `gender`, `dob`, `encrypted_password`, `wca_id`, `delegate_status`, `confirmed_at`, `created_at`, `updated_at`) VALUES ('Tabitha Schultz DDS', 'narciso_bednar@gleasonkozey.org', 'US', 'm', '1966-04-04', '$2a$04$lOPKlmhwZfppTx1ZogBcaOH/AABvRtGaxTiYvdhabT6YlTWjXuuxy', '2003FLEI04', 'delegate', '2016-05-17 03:02:36', '2016-05-17 03:02:36', '2016-05-17 03:02:36')
   (4.0ms)  COMMIT
   (0.1ms)  BEGIN
  Competition Exists (0.3ms)  SELECT  1 AS one FROM `Competitions` WHERE `Competitions`.`id` = BINARY 'FooComp42015' LIMIT 1
  SQL (0.3ms)  INSERT INTO `Competitions` (`name`, `cityName`, `countryId`, `information`, `eventSpecs`, `venue`, `venueAddress`, `website`, `registration_open`, `registration_close`, `showAtAll`, `year`, `month`, `day`, `endMonth`, `endDay`, `cellName`, `id`) VALUES ('Foo Comp 4 2015', 'San Francisco', 'USA', 'Information!', '333 333oh', 'My backyard', 'My backyard street', 'https://www.worldcubeassociation.org', '2016-05-03 03:01:23', '2016-05-10 03:01:23', 1, 2015, 5, 17, 5, 17, 'Foo Comp 4 2015', 'FooComp42015')
  SQL (0.2ms)  INSERT INTO `competition_delegates` (`competition_id`, `delegate_id`, `created_at`, `updated_at`) VALUES ('FooComp42015', 4, '2016-05-17 03:02:36', '2016-05-17 03:02:36')
  User Load (0.4ms)  SELECT `users`.* FROM `users` INNER JOIN `competition_organizers` ON `users`.`id` = `competition_organizers`.`organizer_id` WHERE `competition_organizers`.`competition_id` = 'FooComp42015'
  SQL (0.2ms)  DELETE FROM `competition_organizers` WHERE `competition_organizers`.`competition_id` = 'FooComp42015' AND (1=1)
  SQL (0.2ms)  DELETE FROM `competition_delegates` WHERE `competition_delegates`.`competition_id` = 'FooComp42015' AND (`competition_delegates`.`delegate_id` != 4)
  SQL (0.1ms)  UPDATE `Results` SET `Results`.`competitionId` = 'FooComp42015' WHERE `Results`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `Preregs` SET `Preregs`.`competitionId` = 'FooComp42015' WHERE `Preregs`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `Scrambles` SET `Scrambles`.`competitionId` = 'FooComp42015' WHERE `Scrambles`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `CompetitionsMedia` SET `CompetitionsMedia`.`competitionId` = 'FooComp42015' WHERE `CompetitionsMedia`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `competition_delegates` SET `competition_delegates`.`competition_id` = 'FooComp42015' WHERE `competition_delegates`.`competition_id` = ''
  SQL (0.2ms)  UPDATE `competition_organizers` SET `competition_organizers`.`competition_id` = 'FooComp42015' WHERE `competition_organizers`.`competition_id` = ''
   (3.7ms)  COMMIT
=> 1
irb(main):006:0> FactoryGirl.create(:competition, :confirmed, :visible).delegates.length
   (0.2ms)  BEGIN
  SQL (0.2ms)  INSERT INTO `Persons` (`id`, `name`, `countryId`, `gender`, `year`, `month`, `day`) VALUES ('2003FLEI05', 'Adeline Zemlak', 'USA', 'm', 1966, 4, 4)
   (11.7ms)  COMMIT
   (0.1ms)  BEGIN
  Person Load (0.3ms)  SELECT  `Persons`.* FROM `Persons` WHERE `Persons`.`id` = '2003FLEI05' LIMIT 1
  User Exists (0.2ms)  SELECT  1 AS one FROM `users` WHERE `users`.`email` = BINARY 'phoebe_dare@satterfield.com' LIMIT 1
  User Load (0.4ms)  SELECT  `users`.* FROM `users` WHERE `users`.`wca_id` = '2003FLEI05' LIMIT 1
  User Load (0.2ms)  SELECT `users`.* FROM `users` WHERE `users`.`wca_id` = '2003FLEI05'
  SQL (0.2ms)  INSERT INTO `users` (`name`, `email`, `country_iso2`, `gender`, `dob`, `encrypted_password`, `wca_id`, `delegate_status`, `confirmed_at`, `created_at`, `updated_at`) VALUES ('Adeline Zemlak', 'phoebe_dare@satterfield.com', 'US', 'm', '1966-04-04', '$2a$04$u8FUJEDfxCOH5KOTPi7Bc.cqweaAae7oPErmYFonw5OfF.Tj1LsZi', '2003FLEI05', 'delegate', '2016-05-17 03:02:41', '2016-05-17 03:02:41', '2016-05-17 03:02:41')
   (9.6ms)  COMMIT
   (0.2ms)  BEGIN
  Competition Exists (0.2ms)  SELECT  1 AS one FROM `Competitions` WHERE `Competitions`.`id` = BINARY 'FooComp52015' LIMIT 1
  SQL (0.2ms)  INSERT INTO `Competitions` (`name`, `cityName`, `countryId`, `information`, `eventSpecs`, `venue`, `venueAddress`, `website`, `registration_open`, `registration_close`, `isConfirmed`, `showAtAll`, `year`, `month`, `day`, `endMonth`, `endDay`, `cellName`, `id`) VALUES ('Foo Comp 5 2015', 'San Francisco', 'USA', 'Information!', '333 333oh', 'My backyard', 'My backyard street', 'https://www.worldcubeassociation.org', '2016-05-03 03:01:23', '2016-05-10 03:01:23', 1, 1, 2015, 5, 17, 5, 17, 'Foo Comp 5 2015', 'FooComp52015')
  SQL (0.1ms)  INSERT INTO `competition_delegates` (`competition_id`, `delegate_id`, `created_at`, `updated_at`) VALUES ('FooComp52015', 5, '2016-05-17 03:02:41', '2016-05-17 03:02:41')
  User Load (0.2ms)  SELECT `users`.* FROM `users` INNER JOIN `competition_organizers` ON `users`.`id` = `competition_organizers`.`organizer_id` WHERE `competition_organizers`.`competition_id` = 'FooComp52015'
  SQL (0.1ms)  DELETE FROM `competition_organizers` WHERE `competition_organizers`.`competition_id` = 'FooComp52015' AND (1=1)
  SQL (0.2ms)  DELETE FROM `competition_delegates` WHERE `competition_delegates`.`competition_id` = 'FooComp52015' AND (`competition_delegates`.`delegate_id` != 5)
  SQL (0.1ms)  UPDATE `Results` SET `Results`.`competitionId` = 'FooComp52015' WHERE `Results`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `Preregs` SET `Preregs`.`competitionId` = 'FooComp52015' WHERE `Preregs`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `Scrambles` SET `Scrambles`.`competitionId` = 'FooComp52015' WHERE `Scrambles`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `CompetitionsMedia` SET `CompetitionsMedia`.`competitionId` = 'FooComp52015' WHERE `CompetitionsMedia`.`competitionId` = ''
  SQL (0.1ms)  UPDATE `competition_delegates` SET `competition_delegates`.`competition_id` = 'FooComp52015' WHERE `competition_delegates`.`competition_id` = ''
  SQL (0.1ms)  UPDATE `competition_organizers` SET `competition_organizers`.`competition_id` = 'FooComp52015' WHERE `competition_organizers`.`competition_id` = ''
   (15.9ms)  COMMIT
=> 1
```

Note how `FactoryGirl.create(:competition, :confirmed, :visible)` only created 1 delegate, not two.